scripts/release_notes.rb

```diff
@@ -68,24 +98,87 @@ def gather_issues(offset = 0)
 
 puts "\n----------[Start of notes]----------\n\n"
 
-puts "### Release notes for #{@current_release_name}"
+def apply_category_mappings(grouped_issues, version_prefix = nil)
+  # Step 1: Handle merge mappings
+  CATEGORY_MAPPINGS[:merge].each do |target_category, source_categories|
+    grouped_issues[target_category] ||= []  # Initialize if doesn't exist
+    source_categories.each do |source_category|
+      if grouped_issues.key?(source_category)
+        # Move issues from source to target category and remove source
+        grouped_issues[target_category] += grouped_issues.delete(source_category)
+      end
+    end
+  end
+
+  # Step 2: Apply project-specific transformations
+  if version_prefix && CATEGORY_MAPPINGS[:project_specific][version_prefix]
+    new_grouped_issues = {}
+    transformations = CATEGORY_MAPPINGS[:project_specific][version_prefix]
+
+    grouped_issues.each do |category, category_issues|
+      new_category = category
+      # Apply each regex pattern transformation for this project
+      transformations.each do |pattern, replacement|
+        new_category = new_category.gsub(pattern, replacement)
+      end
+      new_grouped_issues[new_category] = category_issues
+    end
+    grouped_issues = new_grouped_issues
+  end
+
+  # Step 3: Apply general rename mappings
+  new_grouped_issues = {}
+  grouped_issues.each do |category, category_issues|
+    new_category = category
+    CATEGORY_MAPPINGS[:rename].each do |pattern, replacement|
+      new_category = new_category.gsub(pattern, replacement)
+    end
+    new_grouped_issues[new_category] = category_issues
+  end
+
+  new_grouped_issues
+end
+
+def generate_release_notes_for_project(project_name, release_name, version_prefix = nil)
+  # Build full version name (e.g., "hammer-cli-3.16.0" instead of just "3.16.0")
+  full_version_name = version_prefix ? "#{version_prefix}#{release_name}" : release_name
+  release_id = get_version_id(project_name, full_version_name)
+  return if release_id.nil?
+
+  issues = gather_issues(project_name, release_id)
+  return if issues.empty?
+
+  # Group issues by "Project Name" or "Project Name - Category" format
+  grouped_issues = issues.group_by { |issue| "#{issue['project']['name']}#{ ' - ' + issue['category']['name'] if issue.key?('category') }" }
+
+  # Apply configurable category mappings
+  grouped_issues = apply_category_mappings(grouped_issues, version_prefix)
 
-grouped_issues = gather_issues.group_by { |issue| "#{issue['project']['name']}#{ ' - ' + issue['category']['name'] if issue.key?('category') }" }
+  # Sort categories alphabetically
+  grouped_issues = Hash[ grouped_issues.sort_by { |key, val| key } ]
```

Comment:
I know this existed before, but this can be done easier in current Ruby (probably couldn't when it was written). The version in foreman-documentation already does this.
```suggestion
  grouped_issues = grouped_issues.sort
```